floris/tools/layout_functions.py

```diff
@@ -1,376 +1,323 @@
+                * 'markersize' (int): Turbine marker size. Defaults to 10.
+                * 'label' (str): Label for the legend. Defaults to None.
 
-    """
-
-    # Build a dataframe of locations and names
-    df_turbine = pd.DataFrame({
-        'x':fi.layout_x,
-        'y':fi.layout_y
-    })
+    Returns:
+        plt.Axes: The axes object used for the plot.
 
-    # Get some info
-    D = fi.floris.farm.rotor_diameters[0]
-    N_turbine = df_turbine.shape[0]
-    turbines = df_turbine.index
+    Raises:
+        IndexError: If any value in `turbine_indices` is an invalid turbine index.
+    """
 
-    # Set some color information
-    if black_and_white:
-        ec_color = 'k'
-    else:
-        ec_color = 'r'
+    # Generate axis, if needed
+    if ax is None:
+        _, ax = plt.subplots()
 
-    # If we're plotting the rotor, offset the label
-    if plot_rotor:
-        label_offset = D/2
+    # If turbine_indices is not none, make sure all elements correspond to real indices
+    if turbine_indices is not None:
+        try:
+            fi.layout_x[turbine_indices]
+        except IndexError:
+            raise IndexError("turbine_indices does not correspond to turbine indices in fi")
     else:
-        label_offset = 0.
-
-    # If turbine names passed in apply them
-    if turbine_names is not None:
-
-        if len(turbine_names) != N_turbine:
-            raise ValueError(
-                "Length of turbine names array must equal number of turbines within fi"
-            )
-
-        df_turbine['turbine_names'] = turbine_names
-
-    elif one_index_turbine:
-        df_turbine['turbine_names'] = list(range(1,N_turbine+1)) # 1-indexed list
-        df_turbine['turbine_names'] = df_turbine['turbine_names'].astype(int)
+        turbine_indices = list(range(len(fi.layout_x)))
+
+    # Generate plotting dictionary
+    default_plotting_dict = {
+        "color": "black",
+        "marker": ".",
+        "markersize": 10,
+        "label": None,
+    }
+    plotting_dict = {**default_plotting_dict, **plotting_dict}
+
+    # Plot
+    ax.plot(
+        fi.layout_x[turbine_indices],
+        fi.layout_y[turbine_indices],
+        linestyle="None",
+        **plotting_dict,
+    )
 
-    else:
+    # Make sure axis set to equal
+    ax.axis("equal")
 
-        df_turbine['turbine_names'] = list(range(N_turbine)) # 0-indexed list
-        df_turbine['turbine_names'] = df_turbine['turbine_names'].astype(int)
+    return ax
 
 
-    # if no axes provided, make one
-    if not ax:
-        fig, ax = plt.subplots(figsize=(7, 7))
+def plot_turbine_labels(
+    fi: FlorisInterface,
+    ax: plt.Axes = None,
+    turbine_names: List[str] = None,
+    turbine_indices: List[int] = None,
+    label_offset: float = None,
+    show_bbox: bool = False,
+    bbox_dict: Dict[str, Any] = {},
+    plotting_dict: Dict[str, Any] = {},
+) -> plt.Axes:
+    """
+    Adds turbine labels to a turbine layout plot.
```

Comment:
Does `plot_turbine_points` need to have been run on the given ax prior to using this?

Reply:
Something has to have been plotted or I think maplotlib doesn't no exactly how to scale.  Plotting rotors, or plotting wake lines also fine